allocator/sparse_flow_network.go

```diff
@@ -324,19 +345,18 @@ func (fs *sparseFlowNetwork) buildCurrentItemArcs(item int, bound int) []pr.Arc
 	return arcs
 }
 
-// buildMemberArc from member |member| to the sink. Its capacity is:
-// - The scaled proportion of the member's slots relative to item slots
-//   (rounded up), or:
-// - If the node height is over-threshold, the capacity is the full number of
-//   member slots.
-// Intuitively, the Member node will resist having more than its fair share of
-// assignments until sufficient pressure builds within the network to indicate
-// that not all assignments can otherwise be made. At this point the member will
-// allow assignments up to its full capacity.
+// buildMemberArc from member `member` to the sink.
 func (fs *sparseFlowNetwork) buildMemberArc(mf *pr.MaxFlow, id pr.NodeID, member int) []pr.Arc {
 	var c = memberAt(fs.Members, member).ItemLimit()
+	// Constrain to the scaled ItemLimit for our portion of the global assignment problem.
+	c = scaleAndRound(c, len(fs.myItems), len(fs.Items))
```

Comment:
Another check for understanding: I think this line addresses my previous question about respecting item limits when treating this as several smaller networks, yeah?

Reply:
That's right. Each sub-problem gets a corresponding fraction of the total member capacity.